product_docs/docs/pgd/5/admin-tpa/upgrading_major_rolling.mdx

```diff
@@ -0,0 +1,537 @@
+
+### Stop Postgres on the target node
+
+If you're not already on the target node, use SSH to connect to the target node (in this case, kaboom), and then run:
+
+```bash
+sudo systemctl stop postgres
+```
+
+This command halts the server on the current node, in this case, kaboom. Your cluster continues running using the other two nodes.
+
+### Install the new Postgres version
+
+While still on the target node, kaboom, install the new version of Postgres (PG16) and the upgrade tool:
+
+```bash
+sudo apt install edb-bdr5-pg16 edb-bdr-utilities
+```
+
+### Initialize Postgres 16
+
+Make a new data directory for the upgraded Postgres, and give the postgres user ownership of the directory:
+
+```bash
+sudo mkdir /opt/postgres/datanew
+sudo chown -R postgres:postgres /opt/postgres/datanew
+```
+
+Then, initialize Postgres 16 in the new directory:
+
+```bash
+sudo -u postgres /usr/lib/postgresql/16/bin/initdb \
+  -D /opt/postgres/datanew \
+  -E UTF8 \
+  --lc-collate=en_US.UTF-8 \
+  --lc-ctype=en_US.UTF-8 \
+  --data-checksums
+```
+
+This command creates a PG16 data directory for configuration, `/opt/postgres/datanew`.
+
+### Copy configuration for Postgres 15 to Postgres 16
+
+The next step copies the existing configuration (PG15) to the new PG16 data directory.
+
+Copy over the `postgresql.conf`, `postgresql.auto.conf`, and `pg_hba.conf` files and the whole `conf.d` directory:
+
+```bash
+sudo -u postgres cp /opt/postgres/data/postgresql.conf /opt/postgres/datanew/
+sudo -u postgres cp /opt/postgres/data/postgresql.auto.conf /opt/postgres/datanew/
+sudo -u postgres cp /opt/postgres/data/pg_hba.conf /opt/postgres/datanew/
+sudo -u postgres cp -r /opt/postgres/data/conf.d/ /opt/postgres/datanew/
+```
+
+### Rename the PGDATA directories
+
+Next, swap the PG15 and PG16 data directories: 
+
+```bash
+sudo mv /opt/postgres/data /opt/postgres/dataold
+sudo mv /opt/postgres/datanew /opt/postgres/data
+```
+
+!!! Important
+If something goes wrong at some point during the procedure, you may want to rollback/revert a node to the older major version. To do this, rename directories again so that the current data directory, `/opt/postgres/data`, becomes `/opt/postgres/datafailed` and the old data directory, `/opt/postgres/dataold`, becomes the current data directory:
+
+```bash
+sudo mv /opt/postgres/data /opt/postgres/datafailed
+sudo mv /opt/postgres/dataold /opt/postgres/data
+```
+
+This rolls back/reverts the node back to the previous major version of Postgres.
+!!!
+
+### Stop all Postgres instances
+
+Although you previously stopped the Postgres service on the target node, kaboom, to be sure it's stopped, run the `stop` command for both instances of Postgres (PG15 and PG16) on
+kaboom:
+
+```bash
+sudo -u postgres /usr/lib/postgresql/15/bin/pg_ctl -D /opt/postgres/dataold stop
+sudo -u postgres /usr/lib/postgresql/16/bin/pg_ctl -D /opt/postgres/data stop
+```
+
+!!! Note
```

Comment:
This is looking a little messy. Wonder if we can tidy it up a bit. @MattWalliEDB any thoughts on how you'd check servers were stopped without reading the chicken entrails of errors?

Reply:
Can we grep for the pid or something?

Reply:
Or maybe we can just use systemctl status postgresql?

Reply:
I am changing the step to verify that postgres is stopped and using `systemctl status postgres` as it is clear that the server isn't running using this method I believe.